File: examples/Ruby/psenvpub.rb
#!/usr/bin/env ruby

require 'ffi-rzmq'

context = ZMQ::Context.new
publisher = context.socket ZMQ::PUB
publisher.bind "tcp://*:5563"

loop do
  publisher.send_string 'A', ZMQ::SNDMORE
  publisher.send_string "We don't want to see this."

  publisher.send_string 'B', ZMQ::SNDMORE
  publisher.send_string "We would like to see this."

  sleep 1
end

publisher.close

File: examples/Rust/psenvpub.rs
use std::{thread, time};

fn main() {
    let context = zmq::Context::new();
    let publisher = context.socket(zmq::PUB).unwrap();
    assert!(publisher.bind("tcp://*:5563").is_ok());

    loop {
        publisher.send_multipart(["A"], zmq::SNDMORE).unwrap();
        publisher.send("We don't want to see this", 0).unwrap();
        publisher.send_multipart(["B"], zmq::SNDMORE).unwrap();
        publisher.send("We would like to see this", 0).unwrap();
        thread::sleep(time::Duration::from_secs(1));
    }
}
